.github/workflows/build-test.yml

```diff
@@ -70,11 +75,46 @@ jobs:
         run: npx playwright install --with-deps
 
       - name: Run E2E Tests
-        run: npm run test:e2e
+        run: npx playwright test --shard=${{ matrix.shardIndex }}/${{ matrix.shardTotal }}
 
-      - name: Upload Playwright report
+      - name: Upload blob report
         uses: actions/upload-artifact@v4
         if: ${{ !cancelled() }}
+        with:
+          name: blob-report-${{ matrix.shardIndex }}
+          path: blob-report/
+          retention-days: 30
+
+  merge_reports:
```

Comment:
Any particular reason why this is a separate job? Can the merge report step be added after running the e2e tests?

Reply:
This is the recommended process outlined in the official Playwright documentation for [sharding](https://playwright.dev/docs/test-sharding#github-actions-example): running the Playwright tests then merging the reports in the subsequent job.

Reply:
I read the documentation, it's just an example not a recommended process. Also, it's always a good idea to understand how the code we copy work, but I guess it's useless now that it's merged.